src/proto/streams/flow_control.rs

```diff
@@ -146,6 +146,22 @@ impl FlowControl {
         self.window_size -= sz;
     }
 
+    /// Decrement the recv-side window size.
+    ///
+    /// This is called after receiving a SETTINGS ACK frame with a lower
+    /// INITIAL_WINDOW_SIZE value.
+    pub fn dec_recv_window(&mut self, sz: WindowSize) {
+        log::trace!(
+            "dec_recv_window; sz={}; window={}, available={}",
+            sz,
+            self.window_size,
+            self.available
+        );
+        // This should not be able to overflow `window_size` from the bottom.
+        self.window_size -= sz;
+        self.available -= sz;
```

Comment:
I have forgotten a lot about how things work. Isn't it OK for the available window to go in the negative when updating the window size from a settings frame?